### PR TITLE
Move principal exception capture to wrapper

### PR DESCRIPTION
**Pull request by johnsimons**
I've also fixed the cancellation token

### CODE REVIEW

File: source/Server/DirectoryServices/UserPrincipalWrapper.cs [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-using System.Linq;
+using System.Threading;
 
 namespace Octopus.Server.Extensibility.Authentication.DirectoryServices.DirectoryServices
 {
     interface IUserPrincipalWrapper : IPrincipalWrapper, IDisposable
     {
-        IEnumerable<IPrincipalWrapper> GetAuthorizationGroups();
-        IEnumerable<IPrincipalWrapper> GetGroups();
+        IList<IPrincipalWrapper> GetAuthorizationGroups(CancellationToken cancellationToken);
```
Comment (andrewabest) [inline source/Server/DirectoryServices/UserPrincipalWrapper.cs:10]: Should we not still be using `IEnumerable` here?

File: source/Server/DirectoryServices/UserPrincipalWrapper.cs [hunk 2]
```diff
@@ -26,10 +26,40 @@ public void Dispose()
             userPrincipal?.Dispose();
         }
 
-        public IEnumerable<IPrincipalWrapper> GetAuthorizationGroups()
-            => userPrincipal.GetAuthorizationGroups().Select(p => new PrincipalWrapper(p));
+        public IList<IPrincipalWrapper> GetAuthorizationGroups(CancellationToken cancellationToken)
```
Comment (andrewabest) [inline source/Server/DirectoryServices/UserPrincipalWrapper.cs:29]: For consistency we should use `=>` for this declaration

File: source/Server/DirectoryServices/UserPrincipalWrapper.cs (hunk 1 above)
Comment (johnsimons) [reply source/Server/DirectoryServices/UserPrincipalWrapper.cs:10]: I don't see a reason to use IEnumerable

Comment (andrewabest) [reply source/Server/DirectoryServices/UserPrincipalWrapper.cs:10]: Just for the more general interface - we aren't using any features of IList.

**Pull request merged**